src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -121,19 +170,29 @@ EmberAfNetworkCommissioningError OnAddWiFiNetworkCommandCallbackInternal(app::Co
     {
         if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
         {
-            // Assume they are null terminated, see notes above.
-            size_t ssidLen        = strlen(reinterpret_cast<const char *>(ssid));
-            size_t credentialsLen = strlen(reinterpret_cast<const char *>(credentials));
+            VerifyOrExit(ssidLen <= sizeof(sNetworks[i].mData.mWiFi.mSSID),
+                         err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            memcpy(sNetworks[i].mData.mWiFi.mSSID, ssid, ssidLen);
 
-            VerifyOrExit(ssidLen <= kMaxWiFiSSIDLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED);
-            VerifyOrExit(credentialsLen <= kMaxWiFiCredentialsLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED);
+            VerifyOrExit(chip::CanCastTo<std::remove_cv_t<decltype(sNetworks[i].mData.mWiFi.mSSIDLen)>>(ssidLen),
+                         err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            sNetworks[i].mData.mWiFi.mSSIDLen = static_cast<std::remove_cv_t<decltype(sNetworks[i].mData.mWiFi.mSSIDLen)>>(ssidLen);
 
-            memcpy(sNetworks[i].mData.mWiFi.mSSID, ssid, ssidLen);
-            sNetworks[i].mData.mWiFi.mSSIDLen = ssidLen;
+            VerifyOrExit(credentialsLen <= sizeof(sNetworks[i].mData.mWiFi.mCredentials),
+                         err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
             memcpy(sNetworks[i].mData.mWiFi.mCredentials, credentials, credentialsLen);
-            sNetworks[i].mData.mWiFi.mCredentialsLen = credentialsLen;
+
+            VerifyOrExit(chip::CanCastTo<std::remove_cv_t<decltype(sNetworks[i].mData.mWiFi.mCredentialsLen)>>(ssidLen),
+                         err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            sNetworks[i].mData.mWiFi.mCredentialsLen =
+                static_cast<std::remove_cv_t<decltype(sNetworks[i].mData.mWiFi.mCredentialsLen)>>(credentialsLen);
```

Comment:
Again, this feels like it could be a lot more readable without the `remove_cv_t` and with some `using`.